File: WorldFeed/Client/src/app/core/Public/Iterators/IMultiLevelCursorIteratorPage.ts
import {ICursorPageResult} from "../../Core/Iterators/Models/CursorPageResult";

// Iterator page containing results from multiple endpoints when using MultiLevelCursorIterator
export interface IMultiLevelCursorIteratorPage<TParent, TItem, TCursor> extends ICursorPageResult<TItem, TCursor> {
  // Parent items that were transformed into the items
  associatedParentItems: TParent[];
}

File: Sharebook/Server/src/Sharebook.Common/Public/Iterators/IMultiLevelCursorIteratorPage.cs
namespace Sharebook.Common.Public.Iterators
{
    using Sharebook.Common.Iterators.Models;

    /// <summary>
    /// Iterator page containing results from multiple endpoints when using MultiLevelCursorIterator
    /// </summary>
    public interface IMultiLevelCursorIteratorPage<TParent, TItem, TCursor> : ICursorPageResult<TItem, TCursor>
    {
        /// <summary>
        /// Parent items that were transformed into the items
        /// </summary>
        TParent[] AssociatedParentItems { get; set; }
    }
}
